enaml/core/import_hooks.py

```diff
@@ -368,26 +423,203 @@ def get_code(self):
+        else:
+            leaf = fullname + os.path.extsep + 'enaml'
+            for stem in sys.path:
+                enaml_path = os.path.join(stem, leaf)
+                file_info = make_file_info(enaml_path)
+                # To check if cache file is in zip file
+                cache_path = os.path.relpath(file_info.cache_path,
+                                             stem).replace("\\","/")
+                if (cls._is_supported(stem) and os.path.exists(stem)):
+                    try:
+                        with ZipFile(stem,'r') as archive:
+                            name_list = archive.namelist()
+                            if ((leaf in name_list) or
+                                    (cache_path in name_list)):
+                                return cls(file_info,stem)
+                    except IOError:
+                        return
+
+    @classmethod
+    def _is_supported(cls, archive_path):
+        """ Checks if the given archive path is of one of the supported
+            archive types. 
+            
+            Paramters
+            ---------
+            archive_path : string
+                The fully path to the archive.
+
+        Returns
+        -------
+        results : bool
+                Whether the archive is supported or not.
+        """
+        return os.path.splitext(archive_path)[-1].lower() in cls.supported_archives
+
+    def __init__(self, file_info, archive_path):
+        """ Initialize an importer object.
+
+        Parameters
+        ----------
+        file_info : EnamlFileInfo
+            An instance of EnamlFileInfo.
+        archive_path : String 
+            File path to the archive to import from.
+
+        """
+        super(EnamlZipImporter, self).__init__(file_info)
+        self.archive_path = archive_path
+        self.archive = None  # Reference to opened archive
+
+    def get_src_mod_time(self, src_path):
+        """ Overridden to read the modified time of the archive 
+        instead of the source file.
+        """
+        return super(EnamlZipImporter, self).get_src_mod_time(self.archive_path)
+
+    def read_source(self, src_path):
+        """ Overridden to read the source from the currently opened archive 
+        instead of the source file. The `self.archive` must be a reference
+        to the current archive object.
+        """
+        if sys.version_info.major > 2:
+            return self.archive.read(src_path).decode()
+        else:
+            return self.archive.read(src_path)
+
+    def get_code(self):
+        """ Loads and returns the code object for the Enaml module and
+        the full path to the module for use as the __file__ attribute
+        of the module.
+
+        Returns
+        -------
+        result : (code, path)
+            The Python code object for the .enaml module, and the full
+            path to the module as a string.
+
+        """
+
+        # Try to load from the cache on disk
```

Comment:
I am not sure to follow in which case this can actually apply. Could you explain it to me ?